app/src/main/java/com/atomczak/kursakademiaandroida/core/di/AppModule.kt

```diff
@@ -0,0 +1,21 @@
+package com.atomczak.kursakademiaandroida.core.di
+
+import androidx.recyclerview.widget.DividerItemDecoration
+import androidx.recyclerview.widget.GridLayoutManager
+import androidx.recyclerview.widget.LinearLayoutManager
+import org.koin.android.ext.koin.androidContext
+import org.koin.dsl.module
+
+val appModule = module {
```

Comment:
Tutaj sugerowałbym użycie `factory` zamiast `single`. Powody są dwa:

1. Jeśli widok nie będzie już korzystał z tych obiektów to garbage collector je wyczyści i zwolni tym samym trochę pamięci. W przypadku `single` obiekt będzie przechowywany w pamięci przez cały czas działania aplikacji.
2. Może się zdarzyć, że tego typu obiekty będą przechowywały w sobie jakieś wartości, które dla jednego ekranu będą poprawne, a dla drugiego już nie. Z tego względu zawsze lepiej stworzyć oddzielne instancje obiektu dedykowane pod konkretne ekrany.

Z tych powodów bezpiecznie jest zawsze domyślnie używać `factory` do deklarowania zależności.